covsirphy/analysis/scenario.py

```diff
@@ -995,7 +999,115 @@ def score(self, metrics="RMSLE", variables=None, phases=None, past_days=None, na
         return tracker.score(
             metrics=metrics, variables=variables, phases=phases, y0_dict=y0_dict)
 
-    def _fit_create_data(self, oxcgrt_data, model, name):
+    def ensure_records_confirmed(self, target):
+        """
+        Checks for existance of the Confirmed or New Confirmed column.
+        Args:
+            target: one of the target columns in ["Infected", "Confirmed", 
+            "Fatal", "Recovered"].
+        Returns:
+            Either an added New Confirmed column or cleaned records dataframe.
+        """
+        records = self.records(show_figure=False)
+        if not target in records.columns: 
+            print("Confirmed cases not found in records, target is set to Infecteds")
+            target = "Infected"
+        elif target == "New Confirmed":
+            records["New Confirmed"] = records["Confirmed"].diff()
+        
+        records.replace(0, np.nan, inplace=True)
+        records.fillna(method="bfill", inplace=True)
+        records.fillna(method="ffill", inplace=True)
+
+        if len(records) < 1:
+            print("No records for this country")
+            return False
+
+        return records
+
+
+    def estimate_delay(self, oxcgrt_data, indicator="Stringency_index", 
+                        target="Infected", value_range=(7, 35)):
+        """
+
+        Estimate the average delay (in days) between a policy measure taken
+        and the moment of change in for the given target value, for example
+        `Infected`.The change over time is calculated with penalized change
+        point analysis. More information about the method of change point
+        detection can be found on:
+        https://ctruong.perso.math.cnrs.fr/ruptures-docs/build/html/detection/pelt.html.
+        Very long periods are taken out, using df.quantile(0.99).
+
+        Args:
+            oxcgrt_data (covsirphy.OxCGRTData): OxCGRT dataset
+            indicator (str): column variable from the OxCGRT dataset.
+            target (str): one of the target columns to detect changes in,
+                options are["Infected", "Confirmed", "Fatal", "Recovered"].
+                Uses self.records(). 
+            value_range (int, int): tuple, giving the minimum and maximum
+            range to search for change over time. It is assumed that when a
+            policy measure is taken, change will occur after at least 7 days
+            and before 35 days (=default).
+        Returns:
+            days: The estimated number of days of delay for a given country.
+            df: dataframe of target, indicator and calculated periods of change (in days).
+        """
```

Comment:
As "Note", please explain that average recovered period of JHU dataset will be used as returned value when the estimated value was not in `value_range`.